components/supervisor/pkg/ports/ports.go

```diff
@@ -321,16 +328,21 @@ func (pm *Manager) nextState() map[uint32]*managedPort {
 			public = exists && config.Visibility != "private"
 		}
 
-		err := pm.E.Expose(ctx, mp.LocalhostPort, mp.GlobalPort, public)
-		if err != nil {
-			log.WithError(err).WithField("port", *mp).Warn("cannot auto-expose port")
-			continue
-		}
-		log.WithField("port", *mp).Warn("auto-expose port")
+		pm.autoExpose(ctx, mp, public)
 	}
 	return state
 }
 
+func (pm *Manager) autoExpose(ctx context.Context, mp *managedPort, public bool) {
+	err := pm.E.Expose(ctx, mp.LocalhostPort, mp.GlobalPort, public)
+	if err != nil {
+		log.WithError(err).WithField("port", *mp).Warn("cannot auto-expose port")
+		return
+	}
+	pm.autoExposed[mp.LocalhostPort] = mp.GlobalPort
+	log.WithField("port", *mp).Warn("auto-expose port")
```

Comment:
should be Debug level

Reply:
would be Info fine? I wanted to see whether we are spamming or not